src/test/java/org/candlepin/async/tasks/EntitlerJobTest.java

```diff
@@ -231,9 +230,8 @@ private void stubEntitlerErrorResult(String poolId) throws EntitlementRefusedExc
 
     private void stubPoolCuratorPoolsById() {
         final Pool pool = createPool("hello");
-        final CandlepinQuery<Pool> cqmock = mock(CandlepinQuery.class);
-        when(cqmock.iterator()).thenReturn(Collections.singletonList(pool).iterator());
-        when(poolCurator.listAllByIds(anySet())).thenReturn(cqmock);
+        final List<Pool> pools = Collections.singletonList(pool);
```

Comment:
Use `List.of(...)` instead of `Collections.singletonList(...)`

Note that this occurs multiple times throughout this PR.